assets/js/components/KeyMetrics/MetricTileTable.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import PreviewBlock from '../PreviewBlock';
+
+export default function MetricTileTable( props ) {
+	const {
+		Widget,
+		loading,
+		title,
+		rows,
+		columns,
+		limit,
+		zeroState: ZeroState,
+	} = props;
+
+	let tbody = null;
+
+	if ( rows?.length > 0 ) {
+		tbody = rows
+			.slice( 0, limit || rows.length )
+			.map( ( row, rowIndex ) => (
+				<tr key={ rowIndex }>
+					{ columns.map(
+						(
+							{ Component, field, className: columnClassName },
+							colIndex
+						) => {
+							const fieldValue =
+								field !== undefined
+									? get( row, field )
+									: undefined;
+
+							return (
+								<td
+									key={ colIndex }
+									className={ columnClassName }
+								>
+									{ Component && (
+										<Component
+											row={ row }
+											fieldValue={ fieldValue }
+										/>
+									) }
+									{ ! Component && fieldValue }
+								</td>
+							);
+						}
+					) }
+				</tr>
+			) );
+	} else if ( !! ZeroState ) {
+		tbody = (
+			<tr className="googlesitekit-table__body-row googlesitekit-table__body-row--no-data">
+				<td
+					className="googlesitekit-table__body-item"
+					colSpan={ columns.length }
```

Comment:
This `colSpan` doesn't appear to be needed:

```suggestion
```

Reply:
Yep, good catch. Updated.